### ISSUE TITLE
Custom keyboard defaults for languages

### ISSUE THREAD

**Issue by ShalokShalom** [mgba-emu/mgba#905]
German keyboards suffer from the current default setting. In order to provide sane defaults for them, implement "custom keybindings for the different language settings"

**Comment by endrift** [comment]
@rootfather Any input on this? I can try to provide an interface for it.

**Comment by ShalokShalom** [comment]
^-^

**Comment by rootfather** [comment]
@endrift indeed, the default keyboard layout is not optimized for German keyboards with a standard QWERTZ keyboard.

As an example, take the default mapping for the A and B button. Per default, the "B" button is mapped to the key Z and the A button is mapped to X. On a QWERTY keyboard, this works perfectly fine and is a good default setting. However, on a German keyboard that uses the QWERTZ layout, the "Y" and the "Z" keys are swapped compared to QWERTY which renders the default "A" and "B" mapping quite useless.

I agree with @ShalokShalom that an interface which provides custom defaults for the different keyboard layouts out there would be a nice addition.

**Comment by depp** [comment]
I also experience this issue. Looks like #393 and #2020 are related. My thoughts:

* Fixing this issue is easy with the SDL front-end. SDL already provides a platform-independent way to map keys by keyboard location rather than character.
* Qt provides `QKeyEvent::nativeScanCode()` that provides the information we want.
* Unfortunately, Qt's `nativeScanCode()` is platform-specific.
* The scancodes for various platforms are known. SDL has tables, for example. It would require some work.
* To display the input configuration, you'd want to map the configured scancodes to key names, which is not straightforward. Qt does not provide this. There are some platform-specific techniques.
* It would be nice if this didn't break people's existing configurations.
